### ISSUE TITLE
Messages duplicate

### ISSUE THREAD

**Issue by Esshahn** [LucasBassetti/react-simple-chatbot#346]
**Describe the bug**
Messages are doubles with each new trigger.

**To Reproduce**
1. clean install the repo inside a fresh react app2. 
2. use the code example below3. 

**Expected behavior**
messages should appear only once, as defined in the code.
Instead, they appear multiple times (see screenshot)

**Desktop (please complete the following information):**
 - OS: Mac OS Moneterey
 - Browser Latest Chrome, Safari

![Bildschirmfoto 2022-06-28 um 15 51 27](https://user-images.githubusercontent.com/434355/176197087-d2908bd2-39a0-4181-a343-87776693bfa3.png)

**Code**
````
import "./App.css";
import ChatBot from "react-simple-chatbot";

function App() {
  const steps = [
    {
      id: "0",
      message: "Message 0",
      trigger: "1",
    },
    {
      id: "1",
      message: "Message 1",
      trigger: "2",
    },
    {
      id: "2",
      message: "Message 2",
      end: true,
    },
  ];

  return (
    <div className="App">
      <ChatBot steps={steps} />
    </div>
  );
}

export default App;````

**Comment by Darpan-favfly** [comment]
Platform: - Next.js 
File:- next.config.js

**#Issue Solve...**
Solution:- Change reactStrictMode value **TRUE** to **FALSE**

**_======= Code =======_**

```
/** @type {import('next').NextConfig} */
const nextConfig = {
  reactStrictMode: false,
  swcMinify: true,
};

module.exports = nextConfig;
```

**Comment by Eduardo-Miguel** [comment]
> I've investigated more an found the difference that made it work for me. Before I used:
> 
> ```
> import ReactDOM from "react-dom/client";
> ReactDOM.createRoot(document.getElementById('root')).render(
>   <React.StrictMode>
>     <App />
>   </React.StrictMode>
> )
> ```
> 
> which produced the error, switching it to
> 
> ```
> import { render } from "react-dom";
> render(<App />, document.getElementById("root"));
> ```
> 
> made it work. Hope that helps.

This work for me thanks!!!

**Comment by blacky-yg** [comment]
> I've investigated more an found the difference that made it work for me. Before I used:
> 
> ```
> import ReactDOM from "react-dom/client";
> ReactDOM.createRoot(document.getElementById('root')).render(
>   <React.StrictMode>
>     <App />
>   </React.StrictMode>
> )
> ```
> 
> which produced the error, switching it to
> 
> ```
> import { render } from "react-dom";
> render(<App />, document.getElementById("root"));
> ```
> 
> made it work. Hope that helps.

This works for me as well. Thanks

**Comment by IbrahimLakhzine** [comment]
> I've investigated more an found the difference that made it work for me. Before I used:
> 
> ```
> import ReactDOM from "react-dom/client";
> ReactDOM.createRoot(document.getElementById('root')).render(
>   <React.StrictMode>
>     <App />
>   </React.StrictMode>
> )
> ```
> 
> which produced the error, switching it to
> 
> ```
> import { render } from "react-dom";
> render(<App />, document.getElementById("root"));
> ```
> 
> made it work. Hope that helps.

this worked for me too thanks for the help.

**Comment by sajid1545** [comment]
> I've investigated more an found the difference that made it work for me. Before I used:
> 
> ```
> import ReactDOM from "react-dom/client";
> ReactDOM.createRoot(document.getElementById('root')).render(
>   <React.StrictMode>
>     <App />
>   </React.StrictMode>
> )
> ```
> 
> which produced the error, switching it to
> 
> ```
> import { render } from "react-dom";
> render(<App />, document.getElementById("root"));
> ```
> 
> made it work. Hope that helps.

It worked for me too

**Comment by SaleemMalik632** [comment]
i got the same  issue 
 [
        { id: '1', message: 'Welcome to InfoPRO', trigger: '2' },
        { id: '2', message: 'Hi! Guest. Thanks for your interest. In case we get disconnected, Please provide your email address formore', trigger: 'Checkemail' },
        {
            id: 'Checkemail', user: true,
            validator: (value) => {
                if (!value.match(/^[\w-]+(\.[\w-]+)*@([\w-]+\.)+[a-zA-Z]{2,7}$/)) {
                    return 'Please provide a valid email address.';
                }
                return true;
            },
            trigger: 'mailthanks',
        },
        {
            id: 'mailthanks',
            message: 'Thanks! ',
            trigger: 'option',
        },
        {
            id: 'option',
            message: 'Choose the Options',
            end: true
        },
  ]

**Comment by bharghu** [comment]
> Platform: - Next.js File:- next.config.js
> 
> **#Issue Solve...** Solution:- Change reactStrictMode value **TRUE** to **FALSE**
> 
> **_======= Code =======_**
> 
> ```
> /** @type {import('next').NextConfig} */
> const nextConfig = {
>   reactStrictMode: false,
>   swcMinify: true,
> };
> 
> module.exports = nextConfig;
> ```

yes its work fine, thanks to resolve this doubling issue - see the example of implementation - https://shitalacademy.com

**Comment by EliusCaleb** [comment]
> > Platform: - Next.js File:- next.config.js
> > **#Issue Solve...** Solution:- Change reactStrictMode value **TRUE** to **FALSE**
> > **_======= Code =======_**
> > ```
> > /** @type {import('next').NextConfig} */
> > const nextConfig = {
> >   reactStrictMode: false,
> >   swcMinifs y: true,
> > };
> > 
> > module.exports = nextConfig;
> > ```
> 
> yes its work fine, thanks to resolve this doubling issue - see the example of implementation - https://shitalacademy.com

how to you style this component

**Comment by bharghu** [comment]
> > > Platform: - Next.js File:- next.config.js
> > > **#Issue Solve...** Solution:- Change reactStrictMode value **TRUE** to **FALSE**
> > > **_======= Code =======_**
> > > ```
> > > /** @type {import('next').NextConfig} */
> > > const nextConfig = {
> > >   reactStrictMode: false,
> > >   swcMinifs y: true,
> > > };
> > > 
> > > module.exports = nextConfig;
> > > ```
> > 
> > 
> > yes its work fine, thanks to resolve this doubling issue - see the example of implementation - https://shitalacademy.com
> 
> how to you style this component

define -> const theme in top and use theme in <themeprovider>

**Comment by EliusCaleb** [comment]
Thanks  Noted

**Comment by thachsteven** [comment]
> Platform: - Next.js File:- next.config.js
> 
> **#Issue Solve...** Solution:- Change reactStrictMode value **TRUE** to **FALSE**
> 
> **_======= Code =======_**
> 
> ```
> /** @type {import('next').NextConfig} */
> const nextConfig = {
>   reactStrictMode: false,
>   swcMinify: true,
> };
> 
> module.exports = nextConfig;
> ```

Thanks for your helping !

**Comment by nick4fake** [comment]
Sorry, so the suggestion is to... use deprecated rendering APIs?
https://react.dev/blog/2022/03/08/react-18-upgrade-guide#updates-to-client-rendering-apis
